crates/uv/tests/sync.rs

```diff
@@ -2328,3 +2328,44 @@ fn transitive_dev() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+/// Check warning message for https://github.com/astral-sh/uv/issues/6998.
+fn sync_scripts_without_build_system() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "foo"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = []
+
+        [project.scripts]
+        entry = "foo:custom_entry"
+        "#,
+    )?;
+
+    let test_script = context.temp_dir.child("src/__init__.py");
+    test_script.write_str(
+        r#"
+        def custom_entry():
+            print!("Hello")
+       "#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.sync(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: Skipping scripts installation because this project is not a package. To install them, consider setting `tool.uv.package = true` or configuring a custom `build-system`.
```

Comment:
We may want a follow-up issue to enumerate which scripts we aren't installing here, e.g. "Skipping installation of script `entry` because..." (depends on parsing the scripts definition)